src/hashrouter.js

```diff
@@ -0,0 +1,51 @@
+let match
+
+// FIXME: I believe the common `match` will cause data inferring issues. Work this out.
```

Comment:
Not necessary. Essentially we have to match the route with only one path from the last `Switch`.